### PR TITLE
4362: Add FAT tests for redirecting traces to stdout

### PR DESCRIPTION
**Pull request by kbaillie**
OpenLiberty Pull Requester,

ATTENTION, READ THIS: Updated 4/11/2018 - Read and understand this completely,
then delete this entire template. If a reviewer or merger sees this template,
they should fail the review or merge.

If this code change is fixing a user-visible bug in previously released code, it MUST
have an associated issue mentioned in the PR text or description. That Issue also
MUST be labelled “release bug”

This directs automation to scrape this fix for inclusion in the next release's
list of bugs fixed.

If this code is NOT for fixing a released bug, for example new function, fixing
a bug in unreleased function, or other improvements that do not affect the
user-space, no label is needed. An issue could or could not be used based
on the developer’s discretion, but do still delete this text block.

For full details, please see this wiki page:

https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions

### CODE REVIEW

File: dev/fattest.simplicity/src/componenttest/topology/impl/LibertyServer.java [hunk 1]
```diff
@@ -4350,6 +4350,38 @@ public RemoteFile getMatchingLogFile(String regex) throws Exception {
...
+    public List<String> findStringsInLogsUsingMark(String regexp, RemoteFile logFile) throws Exception {
+
+        List<String> matches = new ArrayList<String>();
+        LogSearchResult newOffsetAndMatches;
+
+        Long offset = getMarkOffset(logFile.getAbsolutePath());
+        newOffsetAndMatches = LibertyFileManager.findStringsInFile(regexp, logFile, offset);
+        matches.addAll(newOffsetAndMatches.getMatches()); // get the list of matches found
+
+//        List<String> traceLogBaseNames = listDirectoryContents(logsRoot, traceFileNamePrefix);
```
Comment (Channyboy) [inline dev/fattest.simplicity/src/componenttest/topology/impl/LibertyServer.java]: Left behind comments

File: dev/com.ibm.ws.logging.json_fat/publish/servers/com.ibm.ws.logging.json.ConsoleFormatServer/server.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<!--
+    Copyright (c) 2017 IBM Corporation and others.
```
Comment (Channyboy) [inline dev/com.ibm.ws.logging.json_fat/publish/servers/com.ibm.ws.logging.json.ConsoleFormatServer/server.xml]: copyright!

File: dev/com.ibm.ws.logging.json_fat/publish/servers/com.ibm.ws.logging.json.ConsoleFormatServer/bootstrap.properties [hunk 3]
```diff
@@ -0,0 +1,15 @@
+###############################################################################
+# Copyright (c) 2017 IBM Corporation and others.
```
Comment (Channyboy) [inline dev/com.ibm.ws.logging.json_fat/publish/servers/com.ibm.ws.logging.json.ConsoleFormatServer/bootstrap.properties]: copyright!

File: dev/com.ibm.ws.logging.json_fat/publish/files/jsonConsoleJsonMessageStdout.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<!--
+    Copyright (c) 2017 IBM Corporation and others.
```
Comment (Channyboy) [inline dev/com.ibm.ws.logging.json_fat/publish/files/jsonConsoleJsonMessageStdout.xml]: copyright!

File: dev/com.ibm.ws.logging.json_fat/publish/files/basicConsoleBasicMessageStdout.xml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+<!--
+    Copyright (c) 2017 IBM Corporation and others.
```
Comment (Channyboy) [inline dev/com.ibm.ws.logging.json_fat/publish/files/basicConsoleBasicMessageStdout.xml]: copyright!

File: dev/fattest.simplicity/src/componenttest/topology/impl/LibertyServer.java (hunk 1 above)
Comment (Channyboy) [reply dev/fattest.simplicity/src/componenttest/topology/impl/LibertyServer.java]: Also copyright for the top of this file too.. change to "2011, 2018"

**Comment by kbaillie** [comment]
#build

**Comment by kbaillie** [comment]
#build

**Comment by kbaillie** [comment]
#build

**Comment by kbaillie** [comment]
The build failed due to some previously seen errors occurring in unrelated tests.

**Comment by kbaillie** [comment]
Fixes #4362

**Pull request merged**